homeassistant/components/twentemilieu/__init__.py

```diff
@@ -0,0 +1,94 @@
+"""Support for Twente Milieu."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from twentemilieu import TwenteMilieu
+import voluptuous as vol
+
+from homeassistant.components.twentemilieu.const import (
+    CONF_HOUSE_LETTER,
+    CONF_HOUSE_NUMBER,
+    CONF_POST_CODE,
+    DATA_UPDATE,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ID
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+SCAN_INTERVAL = timedelta(seconds=3600)
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_UPDATE = "update"
+SERVICE_SCHEMA = vol.Schema({vol.Optional(CONF_ID): cv.string})
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up the Twente Milieu components."""
+    async def update(call) -> None:
+        """Service call to manually update the data."""
+        unique_id = call.data.get(CONF_ID)
+        if unique_id is not None:
+            twentemilieu = hass.data[DOMAIN].get(unique_id)
+            if twentemilieu is not None:
+                await twentemilieu.update()
+                async_dispatcher_send(hass, DATA_UPDATE, unique_id)
+        else:
+            tasks = []
+            for twentemilieu in hass.data[DOMAIN].values():
+                tasks.append(twentemilieu.update())
+            await asyncio.wait(tasks)
+
+            for unique_id in hass.data[DOMAIN]:
+                async_dispatcher_send(hass, DATA_UPDATE, unique_id)
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_UPDATE, update, schema=SERVICE_SCHEMA
+    )
+
+    async def _interval_update(now=None) -> None:
+        """Update Twente Milieu data."""
+        await hass.services.async_call(DOMAIN, SERVICE_UPDATE)
+
+    async_track_time_interval(hass, _interval_update, SCAN_INTERVAL)
```

Comment:
Would it be better to schedule one update per Milieu id, from `async_setup_entry`?

Reply:
✅ Addressed. Done.